File: 1291. Sequential Digits.java
class Solution {

    public List<Integer> sequentialDigits(
            int low,
            int high
    ) {

        String digits = "123456789";

        List<Integer> result =
                new ArrayList<>();

        int minLength =
                String.valueOf(low).length();

        int maxLength =
                String.valueOf(high).length();

        for (int length = minLength;
             length <= maxLength;
             length++) {

            for (int start = 0;
                 start + length <= 9;
                 start++) {

                int number = Integer.parseInt(
                        digits.substring(
                                start,
                                start + length
                        )
                );

                if (number >= low &&
                    number <= high) {

                    result.add(number);
                }
            }
        }

        return result;
    }
}

File: 1291.Sequential Digits.py
class Solution:
    def sequentialDigits(
        self,
        low: int,
        high: int
    ) -> List[int]:

        digits = "123456789"

        result = []

        min_length = len(str(low))
        max_length = len(str(high))

        for length in range(
            min_length,
            max_length + 1
        ):

            for start in range(
                0,
                10 - length
            ):

                number = int(
                    digits[start:start + length]
                )

                if low <= number <= high:
                    result.append(number)

        return result
